src/Nethermind/Nethermind.JsonRpc.Test/RpcTest.cs

```diff
@@ -63,10 +60,26 @@ private static IJsonRpcService BuildRpcService<T>(T module) where T : class, IRp
         return service;
     }
 
-    public static JsonRpcRequest GetJsonRequest(string method, params string[]? parameters)
+    public static JsonRpcRequest GetJsonRequest(string method, params string?[]? parameters)
     {
-        var doc = JsonDocument.Parse(JsonSerializer.Serialize(parameters?.ToArray()));
-        var request = new JsonRpcRequest()
+        // NOTE: Since parameters are already passed as strings we want to avoid double serialization of JSON objects
+        var sb = new StringBuilder("[");
+        foreach (var param in parameters ?? [])
+        {
+            if (!string.IsNullOrEmpty(param) && param[0] == '{')
+            {
+                sb.Append(param);
+            }
+            else
+            {
+                sb.Append(JsonSerializer.Serialize(param));
+            }
+            sb.Append(',');
+        }
+        sb.Append(']');
+        var doc = JsonDocument.Parse(sb.ToString(), new JsonDocumentOptions { AllowTrailingCommas = true });
```

Comment:
To avoid a double serialization we skip JSON objects (strings starting with "{"). The solution is kind of hacky since we're constructing a JSON array manually by appending strings but it works.

A proper solution would be to accept `params object?[]?` as argument and perform the serialization here with a provided serializer (ie. `EthereumJsonSerializer`). This would avoid the double serialization and would free unit tests from needing to serialize parameters.

I did not implement the "proper" solution since there are +120 tests that use this code passing already serialized parameters and changing them would make this PR too big.

Reply:
Can we try proper version in a PR to this PR?

Reply:
PR: https://github.com/NethermindEth/nethermind/pull/7547